File: CalculatorF/Calculator.fs
namespace CalculatorF

module Calculator =
    type Operation =
        | Plus = 1
        | Minus = 2
        | Divide = 3
        | Multiply = 4

    let Calculate val1 operation val2 (result:outref<int>) =
        let mutable flag = false
        match operation with
        | Operation.Plus -> result <- val1 + val2
        | Operation.Minus -> result <- val1 - val2
        | Operation.Multiply -> result <- val1 * val2
        | Operation.Divide ->
            if val2 = 0 then flag <- true
            else result <- val1 / val2
        flag

File: Calculator/Calculator.cs
namespace Calculator
{
    public static class Calculator
    {
        public enum Operation
        {
            Plus,
            Minus,
            Divide,
            Multiply,
        }

        public static bool Calculate(int val1, Operation operation, int val2, out int result)
        {
            result = 0;
            switch (operation)
            {
                case Calculator.Operation.Plus:
                    result = val1 + val2;
                    break;
                case Calculator.Operation.Minus:
                    result = val1 - val2;
                    break;
                case Calculator.Operation.Multiply:
                    result = val1 * val2;
                    break;
                case Calculator.Operation.Divide:
                    if (val2 == 0)
                    {
                        return true;
                    }
                    result = val1 / val2;
                    break;
            }
            return false;
        }
    }
}
